app/scripts/services/validateRegistrant.js

```diff
@@ -3,44 +3,71 @@
 angular.module('confRegistrationWebApp')
   .service('validateRegistrant', function validateRegistrant() {
 
```

Comment:
I think you can simplify this function. Something like:
```js
var blockVisibleRuleCheck = function (block, registrant) {
    var answers = registrant.answers;
    var ruleOperand = block.content && block.content.ruleoperand ? block.content.ruleoperand : 'AND';

    angular.forEach(block.rules, function (rule) {
        var answer = _.find(answers, { blockId: rule.parentBlockId });
        if (angular.isUndefined(answer) || answer.value === '') {
            if(ruleOperand === 'AND'){
                return false;
            }
        } else {
            if (rule.operator === '=' && answer.value !== rule.value) {
                if(ruleOperand === 'OR'){
                    return true;
                }
            } else if (rule.operator === '!=' && answer.value === rule.value) {
                if(ruleOperand === 'OR'){
                    return true;
                }
            } else if (rule.operator === '>' && answer.value <= rule.value) {
                if(ruleOperand === 'OR'){
                    return true;
                }
            } else if (rule.operator === '<' && answer.value >= rule.value) {
                if(ruleOperand === 'OR'){
                    return true;
                }
            } else {
                if(ruleOperand === 'AND'){
                    return false;
                }
            }
        }
    });
    return ruleOperand === 'AND';
};
```

You could also keep track of `returnValueAND` and `returnValueOR` (or I might call them `all` and `any`) inside one loop but I think returning once you find a condition that you know is going to make the whole thing return true or false is quicker.